ipn/ipnlocal/local.go

```diff
@@ -832,11 +817,24 @@ func (b *LocalBackend) setClientStatus(st controlclient.Status) {
 		prefs.WantRunning = true
 		prefs.LoggedOut = false
 	}
+	if findExitNodeIDLocked(prefs, st.NetMap) {
+		prefsChanged = true
+	}
 	// Prefs will be written out; this is not safe unless locked or cloned.
 	if prefsChanged {
 		b.prefs = prefs.View()
 	}
 	if st.NetMap != nil {
+		b.mu.Unlock() // respect locking rules for tkaSyncIfNeeded
+		if err := b.tkaSyncIfNeeded(st.NetMap); err != nil {
+			b.logf("[v1] TKA sync error: %v", err)
+		}
+		b.mu.Lock()
+
```

Comment:
cool, which invariants might have been invalidated now and needed to be checked again now that we've reacquired the lock? 😬

Reply:
oh, this is just code movement

Reply:
Yeah, but yes we should revisit this and see if we can make the lock ordering better.